lib/hexapdf/type/image.rb

```diff
@@ -254,6 +257,48 @@ def png_chunk(type, data = '')
         [data.length].pack("N") << type << data << [Zlib.crc32(data, Zlib.crc32(type))].pack("N")
       end
 
+      # Writes the image as TIFF to the given IO stream.
+      def write_tiff(io, info)
+        stream = stream_source
+
+        # header
+        io << ['49492A00'].pack('H*')
+
+        # offset to IFD
+        pad_size = stream.length % 2
+        io << [stream.length + 8 + pad_size].pack('L')
+
+        # data
+        io << Filter.string_from_source(stream)
+
+        # pad to word boundary
+        io << [0].pack('C') * pad_size
+
+        decode_params = [self[:DecodeParms]].flatten[0]
+        compression = if decode_params[:K] > 0 # G3-2D
+                        3
+                      elsif decode_params[:K] < 0 # G4-2D
+                        4
+                      else # G3-1D
+                        2
+                      end
+
+        # IFD entries
+        ifd_entries = [
+            [0x0100, 3, 1, info.width], # width
+            [0x0101, 3, 1, info.height], # height
+            [0x0102, 3, 1, info.bits_per_component], # bits per sample
+            [0x0103, 3, 1, compression], # compression
+            [0x0106, 3, 1, decode_params[:BlackIs1] === false ? 1 : 0], # photometric interpretation
+            [0x0111, 4, 1, 8], # strip offsets
+            [0x0115, 3, 1, 1], # sample per pixel
```

Comment:
We can remove this entry since 1 is the default value.

Reply:
In the TIFF specification it is stated that field 0x0116 "RowsPerStrip" is mandatory. Is there a reason you left this entry out?

Reply:
If `0x0115` is not present, macOS preview won't open the file...

Reply:
Preview app doesn't seem to mind missing `0x0116`, though.

Reply:
Since 0x0116 is mandatory, it should be in the resulting file.

As for 0x0115: The correct value is available in `info.components`, so please use this value instead of hard-coding it.